src/main.rs

```diff
@@ -0,0 +1,287 @@
+                        amount: 100,
+                    })],
+                },
+                target_source: true,
+            }),
+            Arc::new(StartRecast {
+                // TODO: maybe id can be inferred
+                action_id: 0,
+                duration: 45000,
+            }),
+        ],
+        ..Default::default()
+    };
+    let true_thrust = actor::Action {
+        id: 1,
+        name: "True Thrust",
+        results: vec![
+            Arc::new(DoDirectDamage {
+                potency: 290,
+                ..Default::default()
+            }),
+            Arc::new(ApplyCombo(1)),
+            Arc::new(StartGcd::default()),
+        ],
+        ..Default::default()
+    };
+    let vorpal_thrust = actor::Action {
+        id: 2,
+        name: "Vorpal Thrust",
+        results: vec![
+            Arc::new(DoDirectDamage {
+                potency: 140,
+                combo_potency: Some(350),
+                combo_action_id: Some(1),
+                ..Default::default()
+            }),
+            Arc::new(StartGcd::default()),
+        ],
+        ..Default::default()
+    };
+    rotation.add(RotationEntry::new(&life_surge));
+    rotation.add(RotationEntry::new(&vorpal_thrust).with_condition(Arc::new(CheckCombo(1))));
+    rotation.add(RotationEntry::new(&true_thrust));
+    actions.add(life_surge);
+    actions.add(true_thrust);
+    actions.add(vorpal_thrust);
+
+    let mut stats = Stats::default();
+    stats.set_base(Stat::PhysicalWeaponDamage, 134);
+    stats.set_base(Stat::Strength, 5435);
+    stats.set_base(Stat::Dexterity, 326);
+    stats.set_base(Stat::Vitality, 6258);
+    stats.set_base(Stat::Intelligence, 206);
+    stats.set_base(Stat::Mind, 339);
+    stats.set_base(Stat::CriticalHitRate, 3543);
+    stats.set_base(Stat::Determination, 2965);
+    stats.set_base(Stat::DirectHitRate, 1620);
+    stats.set_base(Stat::Defense, 8740);
+    stats.set_base(Stat::MagicDefense, 8740);
+    stats.set_base(Stat::AttackPower, 5435);
+    stats.set_base(Stat::SkillSpeed, 1012);
+    stats.set_base(Stat::AttackMagicPotency, 206);
+    stats.set_base(Stat::HealingMagicPotency, 339);
+    stats.set_base(Stat::SpellSpeed, 380);
+    stats.set_base(Stat::Tenacity, 606);
+    stats.set_base(Stat::Piety, 340);
+
+    commands.spawn().insert(SimState::default());
+    commands.spawn_bundle((
+        Job::DRG,
+        actions,
+        rotation,
+        RecastExpirations::default(),
+        Damage::default(),
+        StatusEffects::default(),
+        stats,
+        ActiveCombos::default(),
+    ));
+    commands.spawn_bundle((
+        Target::default(),
+        Job::None,
+        Actions::default(),
+        Rotation::default(),
+        RecastExpirations::default(),
+        Damage::default(),
+        StatusEffects::default(),
+        Stats::default(),
+        ActiveCombos::default(),
+    ));
+}
+
+fn tick(mut sim_state_query: Query<&mut SimState>) {
```

Comment:
Have you seen Bevy `State` abstraction that attempts to do something very similar?

Reply:
Do you mean this? https://bevy-cheatbook.github.io/programming/states.html

Reply:
Yes, precisely. There's also an example here: https://github.com/bevyengine/bevy/blob/latest/examples/ecs/state.rs